common/implementation/base/src/main/java/com/dfsek/terra/config/loaders/RangeLoader.java

```diff
@@ -36,11 +40,26 @@ public class RangeLoader implements TypeLoader<Range> {
     public Range load(@NotNull AnnotatedType type, @NotNull Object o, @NotNull ConfigLoader configLoader, DepthTracker depthTracker)
     throws LoadException {
         if(o instanceof Map) {
-            Map<String, Integer> map = (Map<String, Integer>) o;
-            return new ConstantRange(map.get("min"), map.get("max"));
+            return configLoader.load(new RangeMapTemplate(), new MapConfiguration((Map<String, Object>) o), depthTracker).get();
         } else {
             int h = configLoader.loadType(Integer.class, o, depthTracker);
             return new ConstantRange(h, h + 1);
         }
     }
+    
+    /*
+     * Template needed so keys can be meta annotated, otherwise the loader could just grab keys directly from the object
+     */
+    public static class RangeMapTemplate implements ObjectTemplate<Range> {
```

Comment:
Why is this a subclass, not in its own file, like all other templates?

Reply:
@Astrashh

Reply:
The template should only really be used inside `RangeLoader`, otherwise there would appear to be two distinct loaders for `Range` (in fact it should probably be declared as private not public). The reason the template is wrapped in a `TypeLoader` is that ranges can be defined both as an integer and a map, so the template class is an implementation detail of that.